### PR TITLE
feat: Support unlabeled object annotations via `upload_object_detection_results`

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s)
Fixes KOL-5294 and KOL-5142

### What change does this PR introduce and why?
Supports unlabeled object annotations via `upload_object_detection_results`.
Also adds an alias function `upload_instance_segmentation_results` -> `upload_object_detection_results`

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/_experimental/object_detection/dataset.py [hunk 1]
```diff
@@ -105,14 +89,14 @@ def multiclass_datapoint_metrics(
...
+        matched_inference=[inf for _, inf in object_matches.matched],
         unmatched_ground_truth=unmatched_ground_truth,
-        unmatched_inference=bbox_matches.unmatched_inf,
+        unmatched_inference=object_matches.unmatched_inf,
         Confused=confused,
         count_TP=len(tp),
         count_FP=len(fp),
         count_FN=len(fn),
-        count_Confused=len(confused),
+        count_Confusedo=len(confused),
```
Comment (gordonhart) [inline kolena/_experimental/object_detection/dataset.py]: Revert typo

File: kolena/metrics/_geometry.py [hunk 2]
```diff
@@ -345,12 +339,12 @@ def match_inferences_multiclass(
 
     confused = []
     for gt, inf in confused_matches.matched:
-        if gt.label != inf.label:
+        if gt.label != inf.label:  # type: ignore[union-attr]
```
Comment (gordonhart) [inline kolena/metrics/_geometry.py]: Is there a way to accommodate optional labels without muting mypy?

Comment (consoull) [reply kolena/metrics/_geometry.py]: Guarding with an `assert` works if we prefer that approach

File: tests/integration/_experimental/object_detection/test_dataset.py [hunk 3]
```diff
@@ -40,51 +119,41 @@ def _assert_result_bbox_contains_fields(df_results: pd.DataFrame, columns: List[
...
+@pytest.mark.parametrize(
+    "annotation,gts,infs",
+    [
+        ("labeled_bboxes", gt_labeled_bbox_single, inf_labeled_bbox_single),
+        ("unlabeled_bboxes", gt_unlabeled_bbox_single, inf_unlabeled_bbox_single),
+        ("labeled_polygons", gt_labeled_polygon_single, inf_labeled_polygon_single),
+        ("unlabeled_polygons", gt_unlabeled_polygon_single, inf_unlabeled_polygon_single),
+    ],
+)
+def test__upload_results__single_class(annotation, gts, infs) -> None:
```
Comment (yifanwu-kolena) [inline tests/integration/_experimental/object_detection/test_dataset.py]: nit: missing type hints for the inputs, seems like we have some unintentional mypy checking skip

File: tests/integration/_experimental/object_detection/test_dataset.py [hunk 4]
```diff
@@ -106,48 +175,93 @@ def test__upload_results__single_class() -> None:
...
-def test__upload_results__multiclass() -> None:
-    name = with_test_prefix(f"{__file__}::test__upload_results__multiclass")
+@pytest.mark.parametrize(
+    "annotation,gts,infs",
+    [
+        ("labeled_bboxes", gt_labeled_bbox_multi, inf_labeled_bbox_multi),
+        ("labeled_polygons", gt_labeled_polygon_multi, inf_labeled_polygon_multi),
+    ],
+)
+def test__upload_results__multiclass(annotation, gts, infs) -> None:
```
Comment (yifanwu-kolena) [inline tests/integration/_experimental/object_detection/test_dataset.py]: ditto.

**Review by yifanwu-kolena — APPROVED** [review]

**Pull request merged**